test/bh/mods-test.js

```diff
@@ -0,0 +1,102 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('.mods()', function() {
+  var tmpls;
+  beforeEach(function() {
+    tmpls = compile(function() {
+      block('button').def()(function() {
+        return JSON.stringify(this.mods);
+      });
+      block('button').elem('*').def()(function() {
+        return JSON.stringify(this.mods);
+      });
+    });
+  });
+
+  it('should return empty mods', function() {
+    tmpls.apply({ block: 'button' }).should.equal('{}');
+  });
+
+  it('should return mods', function() {
+    tmpls.apply({ block: 'button', mods: { type: 'button' } })
+      .should.equal('{"type":"button"}');
+  });
+
+  xit('should return elem mods', function() {
+    tmpls.apply({ block: 'button', elem: 'control',
+        elemMods: { type: 'button' } })
```

Comment:
Should we update `this.mods` if we have `this.ctx.elemMods`? If so, we need create issue... Otherwise test templates should expect elemMods too. [Example](http://bem.github.io/bem-xjst/?bemhtml=block(%27button%27).elem(%27*%27).def()(function()%20%7B%0A%20%20%20%20return%20%5B%0A%20%20%20%20%20%20%20%20JSON.stringify(this.mods)%2C%0A%20%20%20%20%20%20%20%20JSON.stringify(this.elemMods)%2C%0A%20%20%20%20%20%20%20%20JSON.stringify(this.ctx)%0A%20%20%20%20%5D.join(%27%2C%20%27)%3B%0A%7D)%3B&bemjson=(%7B%0A%20%20%20%20block%3A%20%27button%27%2C%0A%20%20%20%20elem%3A%20%27control%27%2C%0A%20%20%20%20elemMods%3A%20%7B%20type%3A%20%27button%27%20%7D%0A%7D)%3B).